components/NavBar.vue

```diff
@@ -53,6 +57,11 @@ const currentPlanName = computed(() => {
               Button clicks left: {{ entitlements.subscriptionButtonClicks }}
             </div>
           </div>
+          <div class="flex flex-col md:flex-row mt-3 md:mt-0 md:items-center">
+            <div class="nav-caption">
+              Subscription renewal at: {{ subscriptionRenewAt }}
```

Comment:
This should handle the case where the `subscriptionRenewAt` value is `undefined` (E.g. the call to obtain and format it hasn't returned or the value isn't set). Something like `{{ subscriptionRenewAt ?? 'Unknown' }}`.